infrastructure/src/main/java/org/corfudb/infrastructure/logreplication/infrastructure/CorfuReplicationDiscoveryService.java

```diff
@@ -494,41 +514,155 @@ private void registerToLogReplicationLock() {
      * as source (sender/producer) or sink (receiver).
      */
     private void onLeadershipAcquire() {
-        switch (localClusterDescriptor.getRole()) {
-            case SOURCE:
-                log.info("Start as Source (sender/replicator)");
-                if (replicationManager == null) {
-                    replicationManager = new CorfuReplicationManager(replicationContext, localNodeDescriptor,
-                        remoteSessionToMetadataManagerMap, serverContext.getPluginConfigFilePath(), getCorfuRuntime(),
-                        replicationConfigManager);
-                } else {
-                    // Replication Context contains the topology which
-                    // must be updated if it has changed
-                    replicationManager.setContext(replicationContext);
+        if (replicationManager == null) {
+            replicationManager = new CorfuReplicationManager(replicationContext,
+                    localNodeDescriptor, remoteSessionToMetadataManagerMap, serverContext.getPluginConfigFilePath(),
+                    getCorfuRuntime(), replicationConfigManager);
+        } else {
+            // Replication Context contains the topology which must be updated if it has changed
+            replicationManager.setContext(replicationContext);
+        }
+
+        initReplicationStatusForRemoteClusters(topologyDescriptor.getRemoteSinkClusters().keySet(), true);
```

Comment:
will it be cleaner to move this to the place where we detect that remoteSinkClusters is not empty?  The if-block on L549

Reply:
No, I realized this cannot be within that if block because initReplicationStatusForRemoteClusters needs to be invoked for all remote clusters that is a source/sink to the local cluster.

moving the 2 lines to L549 would mean that only the remote cluster from which we expect to receive a connection, only for them will initReplicationStatusForRemoteClusters be invoked.